repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -268,6 +270,14 @@ do_upgrade() {
         rv=$?
     fi
 
+    if [ "$rv" -ne 0 ]; then
+        # set the upgrade failed flag to prevent the upgrade from running again
+        # when the emergency shell exits and the upgrade.target is restarted
+        [ -d "$(dirname "$NEWROOT$LEAPP_FAILED_FLAG_FILE")" ] || {
+            "$NEWROOT/bin/touch" "$NEWROOT$LEAPP_FAILED_FLAG_FILE"
+        }
```

Comment:
@matejmatuska now, the `touch` command will be called only if it can fail, which does not make sense. the check has been supposed to be used to trigger the creation of the directory when it does not exist. the current solution does not prevent the loop. check the original suggestion for the hint

Reply:
I somehow got it mixed up. It should be alright now. The original suggestion creates the flag file as a directory though, so I changed it to just create the dir.